.github/workflows/ansible-lint.yml

```diff
@@ -0,0 +1,31 @@
+---
+name: ansible-lint
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  ansible-lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the codebase.
+        uses: actions/checkout@v4
+
+      - name: Set up Python 3.
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.x"
+
+      - name: Run ansible-lint
+        uses: ansible/ansible-lint@v25.7.0
+        with:
+          args: "--sarif-file gh-ansible-lint-results.sarif --exclude molecule"
+
+      - name: Upload SARIF file
+        uses: github/codeql-action/upload-sarif@51f77329afa6477de8c49fc9c7046c15b9a4e79d #v3.29.5
```

Comment:
If we want, we can replace the hash with the actual version number. At least GitHub thinks, GitHub-maintained actions are safe. :-D